plugins/mobile.js

```diff
@@ -13,13 +27,28 @@ Plugin to capture navigator.connection.type on browsers that support it
 	if (!connection) {
 		return;
 	}
-	BOOMR.addVar({
-		"mob.ct": connection.type,
-		"mob.bw": connection.bandwidth,
-		"mob.mt": connection.metered
-	});
-
-	if (connection.downlinkMax) {
-		BOOMR.addVar("mob.lm", connection.downlinkMax);
+
+	function setVars() {
+		var k;
+
+		for (k in param_map) {
+			if (k in connection) {
+				// Remove old parameter value from the beacon because new value might be falsey which won't overwrite old value
+				BOOMR.removeVar("mob." + param_map[k]);
+				if (connection[k]) {
+					BOOMR.addVar("mob." + param_map[k], connection[k]);
+				}
+			}
+		}
+	}
+
+	// If connection information changes, we collect the latest values
+	if (connection.addEventListener) {
+		connection.addEventListener("change", function() {
```

Comment:
Any thoughts on how this could be tested?

Reply:
I think you can call `dispatchEvent` to simulate an event firing.